arangod/RestHandler/RestAnalyzerHandler.cpp

```diff
@@ -0,0 +1,504 @@
+          ));
+
+          return;
+        }
+
+        features.add(*feature);
+      }
+    } else {
+      generateError(arangodb::Result( // generate error
+        TRI_ERROR_BAD_PARAMETER, // code
+        "invalid 'features', expecting body to be of the form { name: <string>, type: <string>[, properties: <object>[, features: <string-array>]] }"
+      ));
+
+      return;
+    }
+  }
+
+  // ...........................................................................
+  // end of parameter parsing
+  // ...........................................................................
+
+  if (!IResearchAnalyzerFeature::canUse(name, auth::Level::RW)) {
+    generateError( // generate error
+      arangodb::rest::ResponseCode::FORBIDDEN, // HTTP code
+      TRI_ERROR_FORBIDDEN, // code
+      std::string("insufficient rights while creating analyzer: ") + body.toString() // message
+    );
+
+    return;
+  }
+
+  IResearchAnalyzerFeature::EmplaceResult result;
+  auto res = analyzers.emplace(result, name, type, properties, features);
+
+  if (!res.ok()) {
+    generateError(res);
+
+    return;
+  }
+
+  if (!result.first) {
+    generateError( // generate error
+      arangodb::rest::ResponseCode::BAD, // HTTP code
+      TRI_errno(), // code
+      std::string("failure while creating analyzer: ") + body.toString() // message
+    );
+
+    return;
+  }
+
+  auto pool = result.first;
+  arangodb::velocypack::Builder builder;
+
+  builder.openObject();
+    addStringRef(builder, ANALYZER_NAME_FIELD, pool->name());
+    addStringRef(builder, ANALYZER_TYPE_FIELD, pool->type());
+    addStringRef(builder, ANALYZER_PROPERTIES_FIELD, pool->properties());
+    builder.add( // add features
+      ANALYZER_FEAURES_FIELD, // key
+      arangodb::velocypack::Value(arangodb::velocypack::ValueType::Array) // value
+    );
+
+      for (auto& feature: pool->features()) {
+        TRI_ASSERT(feature); // has to be non-nullptr
+        addStringRef(builder, feature->name());
+      }
+
+    builder.close();
+  builder.close();
+  generateResult( // generate result
+    result.second // new analyzer v.s. existing analyzer
+    ? arangodb::rest::ResponseCode::CREATED : arangodb::rest::ResponseCode::OK,
+    builder.slice() // analyzer definition
+  );
+}
+
+arangodb::RestStatus RestAnalyzerHandler::execute() {
+  auto* analyzers = arangodb::application_features::ApplicationServer::lookupFeature< // find feature
+    IResearchAnalyzerFeature // feature type
+  >();
+
+  if (!analyzers) {
+    generateError( // generate error
+      arangodb::rest::ResponseCode::BAD, // HTTP code
+      TRI_ERROR_INTERNAL, // code
+      std::string("failure to find feature '") + IResearchAnalyzerFeature::name() + "' while executing REST request for: " + ANALYZER_PATH
+    );
+
+    return arangodb::RestStatus::DONE;
+  }
+
+  auto* sysDatabase = arangodb::application_features::ApplicationServer::lookupFeature< // find feature
```

Comment:
I don't believe rest handlers work after a shutdown, I don't think its possible for the SystemDatabaseFeature to not exist

Reply:
if the API allows for it then it is possible
in tests it's definitely possible